src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java

```diff
@@ -251,18 +266,95 @@ private String saveRectanglesOnImage(List<Rectangle> rectangles, String filename
+        double width = rectangle.width + offset * 2;
+        double height = rectangle.getHeight();
+        return new Rectangle((int) (rectangle.getX() - offset), rectangle.y, (int) width, (int) height);
+    }
+
     private boolean isTargetRectangle(Rectangle rectangle, int row, int column) {
-        double yLeftUp = rectangle.getY() + OFFSET;
-        double xLeftUp = rectangle.getX() + OFFSET;
-        double xRightDown = xLeftUp + rectangle.getWidth() - OFFSET;
-        double yRightDown = yLeftUp + rectangle.getHeight() - OFFSET;
+        double yLeftUp = rectangle.getY();
+        double xLeftUp = rectangle.getX();
+        double xRightDown = xLeftUp + rectangle.getWidth();
+        double yRightDown = yLeftUp + rectangle.getHeight();
         return xLeftUp < column && column < xRightDown && yLeftUp < row && row < yRightDown;
     }
+
+    private boolean isTargetRect(Rect rectangle, int row, int column) {
+        double yLeftUp = rectangle.y;
+        double xLeftUp = rectangle.x;
+        double xRightDown = xLeftUp + rectangle.width;
+        double yRightDown = yLeftUp + rectangle.height;
+        return xLeftUp <= column && column <= xRightDown && yLeftUp <= row && row <= yRightDown;
+    }
+
+    private double getOffset(Rectangle rectangle, List<double[]> list) {
+        double rightOffset;
+        double leftOffset;
+        double leftRectangleX = rectangle.getX();
+        double rightRectangeX = rectangle.getX() + rectangle.width;
+        double rightListX = list.get(0)[0];
+        double leftListX = list.get(0)[2];
+        for (double[] doubles : list) {
+            if (rightListX < doubles[0]) {
+                rightListX = doubles[0];
+            }
+            if (leftListX > doubles[2]) {
+                leftListX = doubles[2];
+            }
+        }
+        leftOffset = leftRectangleX > leftListX ? leftRectangleX - leftListX : 0;
+        rightOffset = rightListX > rightRectangeX ? rightListX - rightRectangeX : 0;
+        return Math.max(rightOffset, leftOffset);
+    }
+
+    private List<double[]> findVerticalLinesWithOpenCV(String fileSource) {
+        Mat dst = new Mat();
+        Mat cdst = new Mat();
+        OpenCV.loadLocally();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        Imgproc.Canny(source, dst, ALL_LINES_THRESHOLD1, ALL_LINES_THRESHOLD2, ALL_LINES_APERTURESIZE, false);
+        Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
+        Mat linesP = new Mat();
+        Imgproc.HoughLinesP(dst, linesP, ALL_LINES_RHO, Math.PI, (int) ALL_LINES_THRESHOLD2,
+                RectangleDetector.VERTICAL_LINE_LENGTH, ALL_LINES_MAXLINEGAP);
+        List<double[]> lines = new ArrayList<>();
+        for (int x = 0; x < linesP.rows(); x++) {
+            double[] l = linesP.get(x, 0);
+            lines.add(l);
+        }
+        return lines;
+    }
+
+    private List<Rect> extractListRect(String filesource) {
+        double WIDTH_PERCENT = 0.15;
```

Comment:
TEXT_PARSING_WIDTH_PERCENT? Consider something more informative please. Consider adding image or width of what is it.

Reply:
Done.